onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -0,0 +1,446 @@
+"""Graph building functions for torchscript graph backend."""
+from __future__ import annotations
+
+import collections
+import typing
+import warnings
+from typing import Any, Optional, Sequence, Union
+
+import numpy as np
+import onnx
+import onnx.checker
+import onnx.defs
+import onnx.shape_inference
+import torch
+from beartype import beartype
+from torch.onnx import _type_utils
+
+import onnxscript
+from onnxscript import evaluator
+from onnxscript import tensor as onnxscript_tensor
+
+__all__ = [
+    "TorchScriptTensor",
+    "TorchScriptGraph",
+    "TorchScriptEvaluator",
+]
+
+
+ValidArgumentType = Union["TorchScriptTensor", str, int, float, bool]
+
+# TODO(justinchuby): Build a context manager to handle source information.
+
+
+class TorchScriptTensor(onnxscript_tensor.Tensor):
+    """A onnxscript tensor that wraps a torchscript Value."""
+
+    # TODO(justinchuby): Create setters for attributes so the exporter can
+    # update the name/shape/type information when they are available.
+
+    def __init__(self, value: torch.Value):
+        super().__init__(None)
+        self._value = value
+
+    @property
+    def value(self) -> np.ndarray:
+        return None
+
+    @property
+    def rank(self) -> int | None:
+        value_type = self._value.type()
+        if value_type is None:
+            return None
+        value_type = typing.cast(torch.TensorType, value_type)
+        return value_type.dim()
+
+    @property
+    def shape(self) -> tuple[int | None, ...] | None:
+        value_type = self._value.type()
+        if value_type is None:
+            return None
+        value_type = typing.cast(torch.TensorType, value_type)
+        shape = value_type.varyingSizes()
+        if shape is None:
+            return None
+        return tuple(shape)
+
+    @property
+    def dtype(self):
+        # TODO: Return numpy dtype
+        return _type_utils.JitScalarType.from_value(
+            self._value, default=_type_utils.JitScalarType.UNDEFINED
+        ).dtype()
+
+    @property
+    def onnx_dtype(self):
+        return _type_utils.JitScalarType.from_value(
+            self._value, _type_utils.JitScalarType.UNDEFINED
+        ).onnx_type()
+
+    def symbolic_value(self) -> torch.Value:
+        """The symbolic Value in torch.Graph."""
+        return self._value
+
+
+@beartype
+def _split_args_kwargs_to_input_attr(
+    onnx_func: onnxscript.OnnxFunction, args, kwargs
+) -> tuple[list[ValidArgumentType], dict[str, ValidArgumentType]]:
+    """Split the function args and kwargs to onnx inputs and attributes.
+
+    Args:
+        onnx_func: The onnx function.
+        args: The positional arguments.
```

Comment:
`The positional arguments` doesn't seem carry much information. Is it the `args` passed into `aten::add.default`?

Reply:
Updated: The positional arguments supplied to `onnx_func`.